rllib/env/single_agent_episode.py

```diff
@@ -458,10 +447,12 @@ def add_env_step(
                 )
 
         # Validate our data.
-        try:  # TODO
-            self.validate()
-        except Exception as e:
-            raise e
+        self.validate()
+
+        # Step time stats.
+        self._last_step_time = time.time()
+        if self._start_time is None:
```

Comment:
In which circumstance is there no start time but a last step time?

Reply:
When we continue from a `cut()` episode chunk.

```
episode = SingleAgentEpisode()
episode.add_env_reset(...)

new_chunk = episode.cut()
new_chunk.add_env_step()   # <- here, there would NOT be a start_time
```

You have a point though, if we cut an ongoing episode at the end of one sample() cycle, then update the model (which might take seconds), then continue sampling in the cut episode chunk, the average episode duration gets a little inaccurate, but only slightly, b/c we do measure each chunk's duration and then add these up in `EnvRunner.get_metrics`.